### PR TITLE
Reporting Ad Impression for Banner and  Interstitial.

### PR DESCRIPTION
**Pull request by dineshgupthabavirisetti**

### CODE REVIEW

File: adapters/InMobi/InMobiAdapter/GADMAdapterInMobiBannerAd.m [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+    GADMAdapterInMobiSetTargetingFromAdConfiguration(_bannerAdConfig);
+    NSDictionary<NSString *, id> *requestParameters =
+    GADMAdapterInMobiCreateRequestParametersFromAdConfiguration(_bannerAdConfig);
+    [_adView setExtras:requestParameters];
+    
+    _adView.delegate = self;
+    [_adView load];
+}
+
+- (void)stopBeingDelegate {
```
Comment (kshtgarg21) [inline adapters/InMobi/InMobiAdapter/GADMAdapterInMobiBannerAd.m:140]: remove

File: adapters/InMobi/InMobiAdapter/GADMAdapterInMobiInterstitialAd.m [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+    }
+}
+
+- (void)stopBeingDelegate {
+    _interstitialAd.delegate = nil;
+}
+
+- (BOOL)isBannerAnimationOK:(GADMBannerAnimationType)animType {
+    return [_interstitialAd isReady];
+}
```
Comment (kshtgarg21) [inline adapters/InMobi/InMobiAdapter/GADMAdapterInMobiInterstitialAd.m:124]: remove both

File: adapters/InMobi/InMobiAdapter/GADMAdapterInMobiInterstitialAd.m [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+    [_interstitalAdEventDelegate willPresentFullScreenView];
+}
+
+- (void)interstitialDidPresent:(nonnull IMInterstitial *)interstitial {
+    NSLog(@"<<<< interstitialDidPresent >>>>");
+}
+
+- (void)interstitial:(nonnull IMInterstitial *)interstitial
+didFailToPresentWithError:(IMRequestStatus *)error {
+    _interstitialRenderCompletionHandler(nil,error);
```
Comment (kshtgarg21) [inline adapters/InMobi/InMobiAdapter/GADMAdapterInMobiInterstitialAd.m:149]: fail callback missing

**Review by kshtgarg21 — CHANGES_REQUESTED** [review]
GADMAdapterInMobiDelegateManager *delegateManager =
      GADMAdapterInMobiDelegateManager.sharedInstance;
  [delegateManager removeDelegateForPlacementIdentifier:_placementIdentifier];
  check if needed for banner or int?

File: adapters/InMobi/InMobiAdapter/GADMAdapterInMobiBannerAd.m (hunk 1 above)
Comment (dineshgupthabavirisetti) [reply adapters/InMobi/InMobiAdapter/GADMAdapterInMobiBannerAd.m:140]: removed

File: adapters/InMobi/InMobiAdapter/GADMAdapterInMobiInterstitialAd.m (hunk 2 above)
Comment (dineshgupthabavirisetti) [reply adapters/InMobi/InMobiAdapter/GADMAdapterInMobiInterstitialAd.m:124]: removed

**Review by kshtgarg21 — CHANGES_REQUESTED** [review]

File: adapters/InMobi/InMobiAdapter/GADMAdapterInMobiRewardedAd.m [hunk 4]
```diff
@@ -143,9 +142,12 @@ - (void)requestRewardedAd {
 }
 
 - (void)presentFromViewController:(nonnull UIViewController *)viewController {
-  if ([_rewardedAd isReady]) {
-    [_rewardedAd showFromViewController:viewController];
-  }
+    if ([_rewardedAd isReady]) {
+        [_rewardedAd showFromViewController:viewController];
+    }  else {
+        IMRequestStatus *error = [[IMRequestStatus alloc] initWithDomain:@"com.inmobi.ads.requeststatus" code:kIMStatusCodeInternalError userInfo:@{NSLocalizedDescriptionKey: @"Rewarded ad not ready to be presented"}];
```
Comment (kshtgarg21) [inline adapters/InMobi/InMobiAdapter/GADMAdapterInMobiRewardedAd.m:148]: use nserror or use gam's error

**Review by kshtgarg21 — CHANGES_REQUESTED** [review]

File: adapters/InMobi/InMobiAdapter/GADMAdapterInMobiUtils.m [hunk 5]
```diff
@@ -114,25 +116,32 @@ void GADMAdapterInMobiSetTargetingFromExtras(GADInMobiExtras *_Nullable extras)
...
-    [IMSdk setYearOfBirth:components.year];
-  }
+    if (connector.userGender == kGADGenderMale) {
+        [IMSdk setGender:kIMSDKGenderMale];
+    } else if (connector.userGender == kGADGenderFemale) {
+        [IMSdk setGender:kIMSDKGenderFemale];
+    }
+    
+    if (connector.userBirthday != nil) {
+        NSDateComponents *components = [NSCalendar.currentCalendar
```
[the remark this answers is not in the document]
Comment (dineshgupthabavirisetti) [reply adapters/InMobi/InMobiAdapter/GADMAdapterInMobiUtils.m:126]: No way to fetch it in Configuration flow. In Connector flow also these fields got depicted and no replacement available.

File: adapters/InMobi/InMobiAdapter/GADMAdapterInMobiRewardedAd.m (hunk 4 above)
Comment (dineshgupthabavirisetti) [reply adapters/InMobi/InMobiAdapter/GADMAdapterInMobiRewardedAd.m:148]: Resolved

**Review by kshtgarg21 — APPROVED** [review]

**Pull request merged**